calliope/core/preprocess/model_run.py

```diff
@@ -573,6 +573,9 @@ def generate_model_run(config, debug_comments, applied_overrides, scenario):
     model_run['run'] = config['run']
     model_run['model'] = config['model']
     model_run['group_constraints'] = config.get('group_constraints', {})
+    if model_run['run']['mode'] == 'spores':
```

Comment:
As with `locations.py`, if we can avoid magically generating the group constraint here, it would be ideal. Is it possible to do it in `run.py` in the backend? Could we access the constraint directly, or force users to set up the group constraint in their yaml config if they want to use spores properly? e.g. they could have to give the name of the cost group constraint in the run configuration for spores:
```
run.spores_options: {score_cost_class: 'spores_score', cost_group: 'systemwide_cost_max', slack: 0.1}
```